frontend/src/metabase-lib/parameters/utils/targets.ts

```diff
@@ -41,16 +40,30 @@ export function getParameterTargetField(
   target: ParameterTarget,
   question: Question,
 ) {
-  if (isDimensionTarget(target)) {
-    const query = question.legacyQuery({ useStructuredQuery: true }) as
-      | NativeQuery
-      | StructuredQuery;
-    const metadata = question.metadata();
-    const dimension = Dimension.parseMBQL(target[1], metadata, query);
+  if (!isDimensionTarget(target)) {
+    return null;
+  }
 
+  const fieldRef = target[1];
+  const query = question.query();
+  const metadata = question.metadata();
+
+  // native queries
+  if (isTemplateTagReference(fieldRef)) {
+    const dimension = TemplateTagDimension.parseMBQL(
+      fieldRef,
+      metadata,
+      question.legacyQuery() as NativeQuery,
+    );
     return dimension?.field();
   }
 
+  if (isConcreteFieldReference(fieldRef)) {
```

Comment:
We are still in the MLv1 land, not entirely MLv2 based. But we no longer parse the query. The check is always far more stable and predictable than before.